state-sync/state-sync-v1/src/coordinator.rs

```diff
@@ -1479,8 +1513,8 @@ impl<T: ExecutorProxyTrait, M: MempoolNotificationSender> StateSyncCoordinator<T
     /// * Kick starts the initial sync process (e.g., syncing to a waypoint or target).
     /// * Issues a new request if too much time has passed since the last request was sent.
     fn check_progress(&mut self) -> Result<(), Error> {
-        if self.is_consensus_executing() {
-            return Ok(()); // No need to check progress or issue any requests (consensus is running).
+        if self.is_consensus_executing() || self.read_only_mode {
+            return Ok(()); // No need to check progress or issue any request
         }
```

Comment:
This is similar to the other blurb, except not with errors.  Maybe make this a function, and just check if there's an error?

Might be useful anywhere else that you need this precondition.

Reply:
Yeah, IIRC, this should only be at 2 locations. I'm gonna leave it as-is for now. But if I add any more, I'll refactor things. 😄